test/common.bash

```diff
@@ -59,17 +74,17 @@ multiprocess_ok () {
 }
 
 need_docker () {
-    # Skip test if $CH_TEST_SKIP_DOCKER is true. If argument provided, use
-    # that tag as missing prerequisite sentinel file.
+    # Skip test if $CH_BUILDER is not Docker. If argument provided, use that
+    # tag as missing prerequisite sentinel file.
     pq=${ch_tardir}/${1}.pq_missing
     if [[ $pq ]]; then
         rm -f "$pq"
     fi
-    if [[ $CH_TEST_SKIP_DOCKER ]]; then
+    if [[ $CH_BUILDER != docker ]]; then
         if [[ $pq ]]; then
             touch "$pq"
         fi
-        skip 'Docker not found or user-skipped'
+        skip 'requires Docker'
     fi
 }
 
```

Comment:
We seem to quote variables within if conditionals inconsistently. Here we have:
```bash
if [[ $CH_BUILDER != docker ]];
```

and then later, on line 166, we see `if [[ -z "$CH_BUILDER" ]]`.